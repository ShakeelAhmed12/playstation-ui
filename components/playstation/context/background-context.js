import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const BackgroundProvider = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}