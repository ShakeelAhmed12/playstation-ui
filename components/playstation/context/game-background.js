import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const GameBackground = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}