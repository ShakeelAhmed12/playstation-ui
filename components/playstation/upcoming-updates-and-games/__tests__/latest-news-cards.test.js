import React from 'react';
import renderer from 'react-test-renderer';
import { GameBackground } from '../../context/game-background';

import LatestNewsCards from '../latest-news-cards';

it('Latest news cards renders correctly', () => {
    const tree = renderer
        .create(
            <GameBackground>
                <LatestNewsCards date="27/01/2021" title="News" articleLink="www.google.com" />
            </GameBackground>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});