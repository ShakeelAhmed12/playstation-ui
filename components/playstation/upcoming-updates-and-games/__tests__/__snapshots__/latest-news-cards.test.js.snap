// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Latest news cards renders correctly 1`] = `
<a
  className="latest-news-cards__StyledCard-sc-118gbx-0 kuDWlk"
  href="www.google.com"
  target="_blank"
>
  <span>
    27/01/2021
  </span>
  <h1>
    News
  </h1>
  <p>
    Find out more
  </p>
</a>
`;
